hail/src/main/scala/is/hail/expr/ir/agg/AggregatorState.scala

```diff
@@ -30,15 +30,15 @@ trait AggregatorState {
   def deserialize(codec: BufferSpec): (EmitCodeBuilder, Value[InputBuffer]) => Unit
 
   def deserializeFromBytes(cb: EmitCodeBuilder, t: PBinary, address: Code[Long]): Unit = {
-    val lazyBuffer = kb.getOrDefineLazyField[MemoryBufferWrapper](Code.newInstance[MemoryBufferWrapper](), (this, "buffer"))
+    val lazyBuffer = kb.getOrDefineLazyField[MemoryBufferWrapper](Code.newInstance[MemoryBufferWrapper](), (this, "bufferWrapper"))
```

Comment:
This was a subtle one -- we have to name these separate things because otherwise they'd refer to the same field! The tuple `(this, "buffer")` was used in both.

Reply:
Is this a problem if we call this on multiple AggregatorState objects? Is that a pattern that will ever come up?

Reply:
No, since our compiled code is single-threaded, and these fields do not store persistent state, only a memoized object that could be expensive to construct many times. The problem with the clashing definition here is that the fields have different types.

Reply:
Ah, I see. The field doesn't actually get constructed from the data; it's a container object for the byte array.

Reply:
yep, exactly.